tests/forms_tests/tests/test_forms.py

```diff
@@ -3576,3 +3576,17 @@ class DataForm(Form):
         f = DataForm({'data': 'xyzzy'})
         self.assertTrue(f.is_valid())
         self.assertEqual(f.cleaned_data, {'data': 'xyzzy'})
+
+    def test_choicefield_uses_separate_widget_instances(self):
```

Comment:
I contemplated for a bit whether I should test the behavior with choices, or the underlying implementation of deepcopying; I opted for the first, but maybe it would make sense to test both separately? As in, put this test with choicefields into `forms_tests/field_tests/test_choicefield.py`, and a simple test that verifies that the instance of a widget is not the same into `forms_tests/field_tests/test_base.py`… Or am I overthinking this?